src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -180,11 +183,38 @@ public void onResultNotifiedOld(ProgrammingExerciseParticipation participation)
             // Find out which test cases were executed and calculate the score according to their status and weight.
             // This needs to be done as some test cases might not have been executed.
             result = testCaseService.updateResultFromTestCases(result, programmingExercise, !isSolutionParticipation && !isTemplateParticipation);
+            setProgrammingExerciseResultRated(result);
             resultRepository.save(result);
         }
         return Optional.ofNullable(result);
     }
 
+    /**
+     * A result is rated if:
+     * - there is no buildAndTestAfterDueDate set and the due date has not passed OR
+     * - there is no buildAndTestAfterDueDate set and and the submission type is INSTRUCTOR / TEST
+     * - there is a buildAndTestAfterDueDate set and the buildAndTestAfterDueDate has passed and the submission type is INSTRUCTOR / TEST
+     *
+     * @param result the rest for which to set the rated attribute (mutates the original object!)
```

Comment:
I think there is a typo or what do you mean with rest?

Reply:
Please also mention in the documentation that you assume that the participation and submission is accessible from the passes result object